torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -130,6 +101,8 @@ def __init__(
         ] = OrderedDict()  # device, rank, params
         self._param_rank: Dict[torch.Tensor, int] = {}
         self._partition_parameters: List[List[Dict]] = []
+        self._index_to_param: Dict[int, torch.Tensor] = {}
```

Comment:
if the model is very big, traversing it for each checkpoint can be a little slow, these are just convenience LUT which are cached from call to call